anyone know why a child_process would self terminate when .on('end') ?